cou/steps/analyze.py

```diff
@@ -78,6 +79,32 @@ def _split_control_plane_and_data_plane(
                 control_plane.append(app)
         return control_plane, data_plane
 
+    @classmethod
+    def _move_to_data_plane_if_required(
+        cls, control_plane: list[OpenStackApplication], data_plane: list[OpenStackApplication]
+    ) -> tuple[list[OpenStackApplication], list[OpenStackApplication]]:
+        """Move control plane applications to data plane if any of its units is on data plane.
+
+        :param control_plane: List of control plane applications.
+        :type control_plane:  list[OpenStackApplication]
+        :param data_plane: List of data plane applications.
+        :type data_plane:  list[OpenStackApplication]
+        :return: Control plane and data plane application lists.
+        :rtype: tuple[list[OpenStackApplication], list[OpenStackApplication]]
+        """
+        data_plane_machines = {
+            unit.machine for units in [app.units for app in data_plane] for unit in units
+        }
+
+        apps_to_move = []
+        for app in control_plane:
+            if len([unit for unit in app.units if unit.machine in data_plane_machines]) > 0:
+                apps_to_move.append(app)
+
+        control_plane_apps = [app for app in control_plane if app not in apps_to_move]
+        data_plane_apps = data_plane + apps_to_move
```

Comment:
I think it will be more readable if we moved this logic to previous for loop and remove app from copy of `control_plane` + add app to copy of `data_plane`. Or maybe even create those list from scratch. What do you say?